### PR TITLE
Transistor support for pending_deletion resources

### PR DESCRIPTION
**Pull request by jayeshpise**

### CODE REVIEW

File: src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java [hunk 1]
```diff
@@ -604,6 +605,20 @@ private CmsRfcCI processTouple(CmsCI templatePlatform, CmsCI designPlatform, Cms
 		List<CmsCIRelation> templInternalRels = new ArrayList<CmsCIRelation>();
 		Map<String, Edge> edges = new HashMap<String, Edge>();
 		for (CmsCIRelation templateRel:templateRels) {
+			CmsCI templatePlatformResource = templateRel.getToCi();
+			if (templatePlatformResource.getCiState().equalsIgnoreCase(STATE_PENDING_DELETION)) {
```
Comment (gauravlall) [inline src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java]: Can we use constant first ?

File: src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java [hunk 2]
```diff
@@ -604,6 +605,20 @@ private CmsRfcCI processTouple(CmsCI templatePlatform, CmsCI designPlatform, Cms
...
+				boolean resourcePendingDeletion = true;
```
Comment (gauravlall) [inline src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java]: can we change the var name 'templatePendingForDeletion'

File: src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java [hunk 3]
```diff
@@ -604,6 +605,20 @@ private CmsRfcCI processTouple(CmsCI templatePlatform, CmsCI designPlatform, Cms
...
+				for (CmsCIRelation userDesignRel:userRels) {
+					CmsCI userPlatformResource = userDesignRel.getToCi();
+					if (userPlatformResource.getCiClassName().replaceAll("catalog", "mgmt.manifest")
```
Comment (gauravlall) [inline src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java]: Can we make replace stricter matching 'catalog.'

File: src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java [hunk 4]
```diff
@@ -604,6 +605,20 @@ private CmsRfcCI processTouple(CmsCI templatePlatform, CmsCI designPlatform, Cms
...
+							.equals(templatePlatformResource.getCiClassName())) {//the resource exists in design too
+						resourcePendingDeletion = false;//do not omit the resource because it has design ci too.
+					}
+				}
+				if (resourcePendingDeletion) {
+					continue;	
+				}
```
Comment (gauravlall) [inline src/main/java/com/oneops/transistor/service/ManifestRfcBulkProcessor.java]: Can we have info statement if we find a resource as its not common

File: src/main/java/com/oneops/transistor/service/DesignRfcProcessor.java [hunk 5]
```diff
@@ -445,8 +445,11 @@ private void clonePlatInternalRels(Map<Long,Long> source2design, long releaseId,
 		List<CmsCIRelation> requiresList = new ArrayList<CmsCIRelation>();
 		for (CmsCIRelation rel : tmplRequires) {
 			if (rel.getAttribute("constraint").getDfValue().matches("1..1|1..*")) {
-				rel.setToCi(cmProcessor.getCiById(rel.getToCiId()));
-				requiresList.add(rel);
+				CmsCI component = cmProcessor.getCiById(rel.getToCiId());
+				if (! component.getCiState().equalsIgnoreCase("pending_deletion")) {
```
Comment (gauravlall) [inline src/main/java/com/oneops/transistor/service/DesignRfcProcessor.java]: We have 'pending_deletion' constant.

**Comment by jayeshpise** [comment]
@gauravlall : changes done as per review comments.

**Comment by vitaliyzinchenko** [comment]
this not gonna work:
if (userPlatformResource.getCiClassName().replaceAll("catalog\\.", "mgmt.manifest.")
   .equals(templatePlatformResource.getCiClassName())) {//the resource exists in design too  
   templateResourcePendingDeletion = false;//do not omit the resource because it has design ci too.
}

you can have 2 resources of the same class, i.e. you can have SystemUser and AppUser resources  of class *.oneops.1.User  
And then you delete AppUser from the pack - so it's pending deletion, but this logic will not remove it from manifest

**Comment by jayeshpise** [comment]
@vitaliyzinchenko : You are right. I have changed the approach now to identify manifest-only template resource.

**Pull request merged**